### PR TITLE
✨ add trace spans and metrics for agent-ngt and index-manager

### PR DESCRIPTION
**Pull request by rinx**
### Description:

- add trace spans for several functions that called by daemons
- add metrics about create index execution (agent)
- add `initial_delay_max_duration` that affects to start auto-index daemon after random duration within it.

### Related Issue:

#343

### How Has This Been Tested?:

nothing.

### Environment:

- Golang Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [X] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/ngt/service/ngt.go [hunk 1]
```diff
@@ -201,6 +214,9 @@ func (n *ngt) Start(ctx context.Context) <-chan error {
 			n.lim = n.dur * 2
 		}
 		defer close(ech)
+
+		time.Sleep(n.idelay)
```
Comment (kpango) [inline pkg/agent/ngt/service/ngt.go:218]:
```suggestion
		time.NewTimer(n.idelay)
		select {
		case <-ctx.Done():
				timer.Stop()
				return ctx.Err()
		case <-timer.C:
		}
		timer.Stop()
```

Comment (rinx) [reply pkg/agent/ngt/service/ngt.go:218]: Thanks.
maybe
```suggestion
		timer := time.NewTimer(n.idelay)
		select {
		case <-ctx.Done():
			timer.Stop()
			return ctx.Err()
		case <-timer.C:
		}
		timer.Stop()
```

Comment (kpango) [reply pkg/agent/ngt/service/ngt.go:218]: exactly

**Pull request merged**